src/numeric-value-typing.js

```diff
@@ -0,0 +1,282 @@
+const failure = null;
```

Comment:
Do we need to credit/acknowledge any external sources?

Reply:
I think all of the standards are now referenced in comments above the functions. Outside of that, I don't think there's other external sources? Most of this PR is a partial implementation of `simplify a calculation tree` and the bits of css-typed-om necessary to perform that simplification.